src/vos/evtree.c

```diff
@@ -2320,9 +2453,17 @@ evt_ent_array_fill(struct evt_context *tcx, enum evt_find_opc find_opc,
 		}
 	}
 out:
+	if (rc == 0 && !d_list_empty(&data_loss_list))
+		rc = evt_data_loss_check(&data_loss_list, ent_array);
```

Comment:
If this "data loss" check is done here, it'll affect entry insert and remove as well. I suppose the intention here is to notify app there is an error when app tries to fetch the "corrupted" data, however, this "corrupted" extent definitely shouldn't impede itself being deleted or other entries being inserted.

It looks to me that a simpler way is to carry the "corrupt" flag in evt_entry, and check the flag to see if it's fetching "corrupted" data in evt_find() (after array sort).

Reply:
I think we can add a parameter "for_read" for evt_ent_array_fill(), then we can only check the bad DTX case for evt_find() case.

Reply:
So if we have chance to refresh the patch, then I will update it; otherwise, it will be done in subsequent DTX patch.

Reply:
Sorry, Just realized that for insert and punch cases, the vos_dtx_check_availability() will NOT return -DER_DATA_LOSS for SV/EV record that attached to corrupted DTX, Your concerned issue has been resolved already. @NiuYawei

Reply:
Ok, thanks, but seems evt_delete() is using INTENT_PURGE, I'm not sure if that is overlooked in the patch.